ci/do_ci.sh

```diff
@@ -120,6 +134,11 @@ elif [[ "$1" == "bazel.coverage" ]]; then
   SRCDIR="${GCOVR_DIR}" "${ENVOY_SRCDIR}"/test/run_envoy_bazel_coverage.sh
   rsync -av "${ENVOY_BUILD_DIR}"/bazel-envoy/generated/coverage/ "${ENVOY_COVERAGE_DIR}"
   exit 0
+elif [[ "$1" == "bazel.coverity" ]]; then
+  setup_gcc_toolchain
```

Comment:
Do we want to use Clang here? The medium term goal in Envoy is to move away from GCC dependence.

Reply:
Doesn't coverity use its own compiler? What does this actually do?

Reply:
Coverity uses its own compiler front end to collect all the data necessary for analysis but uses the target compiler for the compilation. Coverity does support clang. Normally, it would be as simple as swapping setup_gcc_toolchain for setup_clang_toolchain and perhaps a call to cov-configure. However, initial testing with clang revealed an issue where Coverity fails to understand a number of the envoy compilation units resulting in an incomplete analysis file. 

It will take some more time to investigate why this is the case. It could be as simple as Coverity Scan does not yet have all the necessary support for Clang 5.

Reply:
I would say just throw in a comment with what you just said and let's ship this?

Reply:
Works for me.